tests/lax_numpy_reducers_test.py

```diff
@@ -866,5 +867,28 @@ def testCumulativeSumBool(self):
                              dtype=jnp.bool_)
     np.testing.assert_array_equal(np.array([[True], [True], [False]]), out)
 
+  def test__promote_integer_dtype(self) -> None:
+    self.assertTrue(_promote_integer_dtype(jnp.int4), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.uint4), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.int2), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.uint2), dtypes.int_)
+
+  @jtu.sample_product(
+    [dtype for dtype in int_dtypes + unsigned_dtypes],
+  )
+  @jtu.ignore_warning(category=NumpyComplexWarning)
+  def test_low_bitwidth_sum(self, dtype) -> None:
+    sum_jit = jax.jit(jnp.sum)
+    arr = jax.random.randint(
+        key=jax.random.PRNGKey(0),
+        shape=(10, 10),
+        minval=0,
+        maxval=2,
+        dtype=jnp.int32,
+    )
+    s_test = sum_jit(arr.astype(dtype)
+    s_ref = sum_jit(arr.astype(jnp.int32))
```

Comment:
Let's use `self._CheckAgainstNumpy` here so as to use NumPy as the baseline, rather than comparing jax to jax.

Reply:
Hopefully I called it properly